pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -600,6 +601,29 @@ public void destroy() {
     return intIterators;
   }
 
+  private IntIterator[] getSortedBitmapIntIteratorsForBytesColumn(String column) {
+    MutableDictionary dictionary = _dictionaryMap.get(column);
+    int numValues = dictionary.length();
+    IntIterator[] intIterators = new IntIterator[numValues];
```

Comment:
why do we need this complex object here?

Reply:
This is to get a docId iterator for each dictId right?